src/components/FriendRequests.tsx

```diff
@@ -0,0 +1,67 @@
+'use client'
+
+import {FC, useState} from "react";
+import {Check, UserPlus, X} from 'lucide-react';
+import axios from "axios";
+import { useRouter } from 'next/navigation';
+
+interface FriendRequestsProps {
+    incomingFriendRequests: FriendRequest[]
+}
+
+type command = 'accept' | 'deny'
+
+const FriendRequests :FC<FriendRequestsProps> =({incomingFriendRequests})=>{
+    const router = useRouter()
+    const [requests, setRequests] = useState<{
+        senderId:string,
+        senderEmail:string
+    }[]>(incomingFriendRequests);
+
+    const apiPost = async (senderId: string, cmd: command) =>{
```

Comment:
This can be encapsulated in a typescript object complete with dependency injection for setRequests.